main_test.go

```diff
@@ -81,16 +85,92 @@ func TestMessageProcessing(t *testing.T) {
 }
 
-func TestMessageQuarantining(t *testing.T) {
-	t.Run("Test bad non JSON message is quarantined", testMessageProcessing(
-		`bad_message`,
-		`bad_message`,
-		cfg.EqReceiptTopic, cfg.EqReceiptProject, cfg.DlqRoutingKey))
+func TestMessageQuarantiningBadJson(t *testing.T) {
+	testMessageQuarantining("bad_message", "Test bad non JSON message is quarantined", t)
+}
 
-	t.Run("Test bad message missing transaction ID is quarantined", testMessageProcessing(
-		`{"thisMessage": "is_missing_tx_id"}`,
-		`{"thisMessage": "is_missing_tx_id"}`,
-		cfg.EqReceiptTopic, cfg.EqReceiptProject, cfg.DlqRoutingKey))
+func TestMessageQuarantiningMissingTxnId(t *testing.T) {
+	testMessageQuarantining(`{"thisMessage": "is_missing_tx_id"}`, "Test bad message missing transaction ID is quarantined", t)
+}
+
+func testMessageQuarantining(messageToSend string, testDescription string, t *testing.T) {
+	var requests []*http.Request
+	var requestBody []byte
+
+	mockResult := "Success!"
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(200)
+		w.Write([]byte(mockResult))
+		requests = append(requests, r)
+		requestBody, _ = ioutil.ReadAll(r.Body)
+	}))
+	defer srv.Close()
+
+	cfg.QuarantineMessageUrl = srv.URL
+
+	t.Run(testDescription, testMessageProcessingQuarantine(
+		messageToSend,
+		cfg.EqReceiptTopic, cfg.EqReceiptProject))
+
+	time.Sleep(1 * time.Second)
+
+	if len(requests) != 1 {
+		t.Errorf("Unexpected number of calls to Exception Manager")
+		return
+	}
+
+	var quarantineBody models.MessageToQuarantine
+	err := json.Unmarshal(requestBody, &quarantineBody)
+	if err != nil {
+		t.Errorf("Could not decode request body sent to Exception Manager")
+		return
+	}
+
+	if assertEqual("application/json", quarantineBody.ContentType, "Dodgy content type", t) {
+		return
+	}
+
+	if assertEqual("Error unmarshalling message", quarantineBody.ExceptionClass, "Dodgy exception class", t) {
+		return
+	}
+
+	if assertEqual(1, len(quarantineBody.Headers), "Dodgy headers", t) {
+		return
+	}
+
+	if len(quarantineBody.Headers["pubSubId"]) == 0 {
+		t.Errorf("Dodgy pubSubId header, expected non-zero length")
+		return
+	}
+
+	if assertEqual(64, len(quarantineBody.MessageHash), "Dodgy message hash", t) {
+		return
+	}
+
+	if assertEqual(messageToSend, string(quarantineBody.MessagePayload), "Dodgy message payload", t) {
+		return
+	}
+
+	if assertEqual(cfg.EqReceiptSubscription, quarantineBody.Queue, "Dodgy quarantine queue", t) {
+		return
+	}
+
+	if assertEqual("none", quarantineBody.RoutingKey, "Dodgy routing key", t) {
+		return
+	}
+
+	if assertEqual("Pubsub Adapter", quarantineBody.Service, "Dodgy routing key", t) {
+		return
+	}
+}
+
+func assertEqual(expected interface{}, actual interface{}, feedback string, t *testing.T) bool {
+	if expected != actual {
+		t.Errorf("%v, expected %v, actual %v", feedback, expected, actual)
+		return true
+	}
+
+	return false
```

Comment:
`assertEqual` returning `false` for equal inputs reads very strangely to me, definitely think it should be inverted